dataflow/src/main/java/org/checkerframework/dataflow/analysis/BackwardAnalysisImpl.java

```diff
@@ -0,0 +1,386 @@
+            FlowRule flowRule,
+            boolean addToWorklistAgain) {
+        if (flowRule != FlowRule.EACH_TO_EACH) {
+            throw new BugInCF(
+                    "backward analysis always propagate EACH to EACH, because there is no control flow.");
+        }
+
+        addStoreAfter(pred, node, currentInput.getRegularStore(), addToWorklistAgain);
+    }
+
+    protected void addStoreAfter(Block pred, Node node, S s, boolean addBlockToWorklist) {
+        // If Block {@code pred} is an ExceptionBlock, decide whether the
+        // block of passing node is an exceptional successor of Block {@code pred}
+        if (pred instanceof ExceptionBlock
+                && (((ExceptionBlock) pred).getSuccessor() == null
+                        || (node != null
+                                && ((ExceptionBlock) pred).getSuccessor().getId()
+                                        != node.getBlock().getId()))) {
+            // If the block of passing node is an exceptional successor of Block @{code pred},
+            // propagating store to the {@code exceptionStores}
+
+            // Currently I don't track the label of an exceptional edge from Exception Block
+            // to it's exceptional successors in backward direction, instead, all exception stores
+            // of exceptional successors of an Exception Block will merge to one exception store
+            // at the Exception Block
+
+            ExceptionBlock ebPred = (ExceptionBlock) pred;
+
+            S exceptionStore = exceptionStores.get(ebPred);
+
+            S newExceptionStore = (exceptionStore != null) ? exceptionStore.leastUpperBound(s) : s;
+            if (!newExceptionStore.equals(exceptionStore)) {
+                exceptionStores.put(ebPred, newExceptionStore);
+                addBlockToWorklist = true;
+            }
+        } else {
+            S predOutStore = getStoreAfter(pred);
+
+            S newPredOutStore = (predOutStore != null) ? predOutStore.leastUpperBound(s) : s;
+
+            if (!newPredOutStore.equals(predOutStore)) {
+                outStores.put(pred, newPredOutStore);
+                inputs.put(pred, new TransferInput<>(node, this, newPredOutStore));
+                addBlockToWorklist = true;
+            }
+        }
+
+        if (addBlockToWorklist) {
+            addToWorklist(pred);
+        }
+    }
+
+    protected S getStoreAfter(Block b) {
+        return readFromStore(outStores, b);
+    }
+
+    @Override
+    public S runAnalysisFor(
+            Node node,
+            boolean before,
+            TransferInput<V, S> transferInput,
+            IdentityHashMap<Node, V> nodeValues,
+            Map<TransferInput<V, S>, IdentityHashMap<Node, TransferResult<V, S>>> analysisCaches) {
+        Block block = node.getBlock();
+        Node oldCurrentNode = currentNode;
+
+        // TODO: why if analysis is running, then the Store of passing node is
+        //  analysis.currentInput.getRegularStore()?
```

Comment:
TODO: Understand why the Store of passing node is blah when the analysis is running